provider/server/exec.go

```diff
@@ -5,11 +5,182 @@ package server
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
 
+	"github.com/creack/pty"
+	"github.com/digitalocean/go-libvirt"
+	"github.com/go-logr/logr"
 	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	remotecommandserver "github.com/ironcore-dev/ironcore/poollet/machinepoollet/iri/streaming/remotecommand"
+	"github.com/ironcore-dev/libvirt-provider/pkg/api"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
+	"github.com/ironcore-dev/libvirt-provider/pkg/store"
+	"github.com/moby/term"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/client-go/tools/remotecommand"
 )
 
+const (
+	StreamCreationTimeout = 30 * time.Second
+	StreamIdleTimeout     = 2 * time.Minute
+)
+
+type executorExec struct {
+	Libvirt         *libvirt.Libvirt
+	ExecRequest     *iri.ExecRequest
+	VirshExecutable string
+	Machine         *api.Machine
+}
+
 func (s *Server) Exec(ctx context.Context, req *iri.ExecRequest) (*iri.ExecResponse, error) {
+	machineID := req.MachineId
+	log := s.loggerFrom(ctx, "MachineID", machineID)
+
```

Comment:
```suggestion
	log := s.loggerFrom(ctx, "MachineID", req.MachineId)
```

variable is used only once, pls remove it or replace others as well